### ISSUE TITLE
Add logging to enrollment page

### ISSUE THREAD

**Issue by trombonekenny** [PrairieLearn/PrairieLearn#1493]
The `enrollments` table doesn't have a history (add_date, delete_date) - only current status in the database, and nothing is put in the `audit_log` for enrollment actions. We should add logging for enrollment actions via one of those methods.

**Comment by mwest1066** [comment]
Absolutely. This is a clear oversight. I guess the best thing to do would be to add a sprocs for `enrollments_insert` and `enrollments_delete`, and then it would be straightforward to add audit data.

Adding an `enrollments.add_date` column would be straightforward.

Adding `enrollments.delete_date` and switching to soft-deletes would be a lot more work, because there are many places in the code where we JOIN over enrollments and we’d need to add a `WHERE delete_date IS NULL` to all of them. The `enrollments` table was designed to support hard deletes without losing any data (we don’t delete any actual user assessment/question data when we delete an enrollment) and it already transparently supports undelete (if a student un-enrolls and re-enrolls, everything is just like it never happened).

My personal feeling is that it’s easier to only audit enrollments via `audit_logs` and don’t add `add_date` or `delete_date` to `enrollments` itself.
